packages/checkout/components/store-notices-container/index.tsx

```diff
@@ -12,37 +17,53 @@ import StoreNotices from './store-notices';
 import SnackbarNotices from './snackbar-notices';
 import type { StoreNoticesContainerProps, StoreNotice } from './types';
 
-const formatNotices = (
-	notices: StoreNotice[],
-	context: string
-): StoreNotice[] => {
+const formatNotices = ( notices: Notice[], context: string ): StoreNotice[] => {
 	return notices.map( ( notice ) => ( {
 		...notice,
 		context,
-	} ) );
+	} ) ) as StoreNotice[];
 };
 
 const StoreNoticesContainer = ( {
 	className = '',
 	context,
 	additionalNotices = [],
 }: StoreNoticesContainerProps ): JSX.Element | null => {
-	const suppressNotices = useSelect( ( select ) =>
-		select( PAYMENT_STORE_KEY ).isExpressPaymentMethodActive()
+	const { suppressNotices, registeredContainers } = useSelect(
+		( select ) => ( {
+			suppressNotices:
+				select( PAYMENT_STORE_KEY ).isExpressPaymentMethodActive(),
+			registeredContainers: select(
+				STORE_NOTICES_STORE_KEY
+			).getRegisteredContainers(),
+		} )
 	);
 
+	// Find sub-contexts that have not been registered. We will show notices from those contexts here too.
+	const allContexts = getNoticeContexts();
+	const subContexts = allContexts.filter(
+		( subContext: string ) =>
+			subContext.includes( context + '/' ) &&
+			! registeredContainers.includes( subContext )
+	);
+
+	// Get notices from the current context and any sub-contexts and append the name of the context to the notice
+	// objects for later reference.
 	const notices = useSelect< StoreNotice[] >( ( select ) => {
 		const { getNotices } = select( 'core/notices' );
 
-		return formatNotices(
-			( getNotices( context ) as StoreNotice[] ).concat(
-				additionalNotices
+		return [
+			...subContexts.map( ( subContext: string ) =>
```

Comment:
Does the first spread result in a `StoreNotice[][]` and the second `StoreNotice[]`, do we need to flatten the first one?